http/batch_index.go

```diff
@@ -0,0 +1,114 @@
+//  Copyright (c) 2016 Couchbase, Inc.
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+package http
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/blevesearch/bleve"
+	"io"
+	"net/http"
+)
+
+type MetaData struct {
+	Action string `json:"action"`
+	Index  string `json:"index"`
+	Id     string `json:"id"`
+}
+
+type BatchIndexHandler struct {
+}
+
+func NewBatchIndexHandler() *BatchIndexHandler {
+	return &BatchIndexHandler{}
+}
+
+func (h *BatchIndexHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	openBatches := map[string]*bleve.Batch{}
+
+	dec := json.NewDecoder(req.Body)
+	isMeta := true
+	var currentMeta MetaData
+
+	for {
+		if isMeta {
+			var meta MetaData
+			if err := dec.Decode(&meta); err == io.EOF {
+				break
+			} else if err != nil {
+				showError(w, req, fmt.Sprintf("error parsing meta JSON: %v", err), 400)
```

Comment:
I generally prefer using the status names exported by the http package, mainly because I forget what the less common numbers mean.  In this case we could use:

http.StatusBadRequest instead of 400